src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/GbfsFreeVehicleStatusMapper.java

```diff
@@ -33,13 +40,14 @@ public VehicleRentalVehicle mapFreeVehicleStatus(GBFSBike vehicle) {
       VehicleRentalVehicle rentalVehicle = new VehicleRentalVehicle();
       rentalVehicle.id = new FeedScopedId(system.systemId, vehicle.getBikeId());
       rentalVehicle.system = system;
-      rentalVehicle.name = new NonLocalizedString(vehicle.getBikeId());
+      rentalVehicle.name = new NonLocalizedString(getName(vehicle));
       rentalVehicle.longitude = vehicle.getLon();
       rentalVehicle.latitude = vehicle.getLat();
       rentalVehicle.vehicleType =
```

Comment:
Avoid resolving null keys on the unmodifiable map:
```suggestion
      rentalVehicle.vehicleType =
        Optional
          .ofNullable(vehicle.getVehicleTypeId())
          .map(t -> vehicleTypes.getOrDefault(t, RentalVehicleType.getDefaultType(system.systemId)))
          .orElseGet(() -> RentalVehicleType.getDefaultType(system.systemId));
```
(and then remove the four lines previously changed/added below.)

Reply:
I would absolutely love to write that sort of code. I argued for extensive use of `Optional` for quite a while, but there is a real resistance to writing that sort of code and a preference for using null checks instead.

I don't understand it either but I'm quite reluctant to re-opening that discussion.

Reply:
I've tried as well. :crying_cat_face: